srtcore/core.cpp

```diff
@@ -2574,7 +2574,7 @@ int CUDT::processSrtMsg_HSREQ(const uint32_t *srtdata, size_t len, uint32_t ts,
         return SRT_CMD_REJECT;
     }
 
-    if (len < SRT_HS_LATENCY + 1)
+    if (bytelen < ((SRT_HS_LATENCY + 1) * sizeof(uint32_t)))
```

Comment:
Please use `SRT_HS_E_SIZE`, not `SRT_HS_LATENCY + 1` (this condition probably comes from times when this symbol wasn't yet introduced, but the use of this symbol clearly states the intention). The intention is to maintain the compatibility with the version as explained near the definition of the `SRT_CMD_HSREQ_MINSZ` symbol. This symbol has a value of 8, which is 2*int. That's the probable reason of this condition. The `SRT_HS_E_SIZE` has the value of 3.

I'd add also at least a command like this:
```
// This condition is due to the value of SRT_CMD_HSREQ_MINSZ, which is less than SRT_HS_E_SIZE*sizeof(int32_t)
```

Reply:
The intent here is to ensure that the Handshake Extension Message is big enough to have the TSBPD fields. At the moment those fields are the last in the extension message, and `SRT_HS_LATENCY + 1` is identical to `SRT_HS_E_SIZE`, but it could change in the future, and the intent is to check the latency is present. Therefore I would vote for `SRT_HS_LATENCY + 1` here.

#### Handshake Extension Message:

```
 0                   1                   2                   3
 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
|                          SRT Version                          |
+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
|                           SRT Flags                           |
+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
|      Receiver TSBPD Delay     |       Sender TSBPD Delay      |
+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
```

Reply:
Updated `SRT_HS_E_SIZE` will require rework of this function anyway because the conditions for backward compatibility functionalities and honoring partial information here will be dramatically changed. And when you do use this symbol, it will be at least easier to find.

I think it would be also reasonable to add a comment that describes better the situation - and precise that the instance versions that might have this are earlier than 1.2.0, so this might likely be removed in the future.

Reply:
How about this way then?
```c++
    SRT_ASSERT(SRT_HS_E_SIZE == (SRT_HS_LATENCY + 1));
    if (bytelen < (SRT_HS_E_SIZE * sizeof(uint32_t)))
    {
        // Handshake extension message includes VERSION, FLAGS and LATENCY
        // (3 x 32 bits). SRT v1.2.0 and earlier might supply shorter extension message,
        // without LATENCY fields.
        // It is acceptable, as long as the latency flags aren not set on our side.
        //
        //  0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
        // +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
        // |                          SRT Version                          |
        // +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
        // |                           SRT Flags                           |
        // +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
        // |      Receiver TSBPD Delay     |       Sender TSBPD Delay      |
        // +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
```

Reply:
Should work.